gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java

```diff
@@ -292,6 +292,20 @@ protected void setTaskStagingAndOutputDirs() {
           new Path(this.jobState.getProp(ConfigurationKeys.TASK_DATA_ROOT_DIR_KEY), this.jobId).toString();
       this.jobState.setProp(ConfigurationKeys.TASK_DATA_ROOT_DIR_KEY, taskDataRootDirWithJobId);
 
+      // Add jobId to writer staging dir
```

Comment:
What if a user only specifies `WRITER_STAGING_DIR` and `WRITER_OUTPUT_DIR`, but not`TASK_DATA_ROOT_DIR_KEY`, will you fix still work?